src/GitHub.Exports/Services/PackageSettings.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using GitHub.VisualStudio;
+using Microsoft.VisualStudio.Settings;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Settings;
+using SettingsStore = GitHub.Helpers.SettingsStore;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace GitHub.Services
+{
+    [Export(typeof(IPackageSettings))]
+    [PartCreationPolicy(CreationPolicy.NonShared)]
+    public class PackageSettings : IPackageSettings
+    {
+        readonly SettingsStore settingsStore;
+
+        readonly Dictionary<string, object> SettingsProperties = new Dictionary<string, object> {
+            { "CollectMetrics", true },
+        };
+
+        Dictionary<string, object> settings = new Dictionary<string, object>();
+
+        public bool CollectMetrics
+        {
+            get
+            {
+                object val;
+                TryGetSetting(nameof(CollectMetrics), out val);
+                return (bool)val;
+            }
+            set { settings[nameof(CollectMetrics)] = value; }
+        }
+
+        [ImportingConstructor]
+        public PackageSettings([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        {
+            var sm = new ShellSettingsManager(serviceProvider);
+            settingsStore = new SettingsStore(sm.GetWritableSettingsStore(SettingsScope.UserSettings), Info.ApplicationInfo.ApplicationSafeName);
+            LoadSettings();
+        }
+
+        public void Save()
+        {
+            SaveSettings();
+        }
+
+        bool TryGetSetting(string name, out object value)
+        {
+            if (!settings.ContainsKey(name))
+            {
+                if (!SettingsProperties.ContainsKey(name))
+                    Debug.Assert(false, "PackageSettings: " + name + " setting missing. Did you add this setting to the SettingsProperties dictionary?");
+                else
+                    Debug.Assert(false, "PackageSettings: " + name + " setting missing. Registry might be corrupt.");
+#if !DEBUG
+                VsOutputLogger.WriteLine("PackageSettings: " + name + " setting missing.");
+#endif
+                value = SettingsProperties[name]; // default value
```

Comment:
Couldn't this throw in non-DEBUG builds? On line 54 we check to see if the `name` is in the `SettingsProperties` dictionary, but we don't bail out if it's not in there.